### DISCUSSION TITLE
[A/L] (2021/22) Funzioni - Esercizio 19

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#259] [Esercizi A-L]
![image](https://user-images.githubusercontent.com/69389461/152316029-3e18c06c-639a-426f-a0fb-f0ea5b9b4844.png)

**Comment by LuginiAndrea** [comment]
Si
$$f: \mathbf{Z} \rightarrow \mathbf{R}, f\left(x\right) = x \times 2$$
Dimostriamone l'iniettività per assurdo
$$x_1 \neq x_2, f\left(x_1\right) = f\left(x_2\right) \Longrightarrow x_1 \times 2 = x_2 \times 2 \Longrightarrow x_1 = x_2$$
$$g: \mathbf{R} \rightarrow \mathbf{R^+}, g\left(z\right) = z^2$$
Dimostriamone la suriettività
$$z^2 = y \Longrightarrow z = \pm \sqrt{y} \Longrightarrow \textrm{ la C.E. della radice sono rispettate in quanto } y \in \mathbf{R^+}$$
$$h: \mathbf{N} \rightarrow \mathbf{R^+}, h= g \circ f$$
La loro composta non è iniettiva: basti prendere in caso $2$ e $-2$
$$\left(-2 \times 2\right)^2 = \left(2 \times 2\right)^2 \Longrightarrow -4^2 = 4^2 \Longrightarrow 16 = 16$$
La funzione si dimostra non essere neanche suriettiva, basti considerare $y = 2 \in \mathbf{R^+}$
$$2 = \left(x \times 2\right)^2 \Longrightarrow \pm \sqrt[2]{2} = x \times 2 \Longrightarrow \nexists x \in \mathbf{Z} \textrm{ t.c. } x \times 2 = \pm \sqrt{2}$$
